src/ex3/scripts/ItemManager.mjs

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable import/extensions */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable array-callback-return */
+/* eslint-disable camelcase */
+import PokemonClient from './PokemonClient.mjs';
+import FileManager from './FileManager.mjs';
+
+export default class ItemManager {
+  constructor() {
+    this.pokemon_client = new PokemonClient();
+    this.pokemons = [];
+    this.tasks = [];
+    this.file_manager = new FileManager();
+  }
+
+  /**
+     * init pokemon names array
+     */
+  async init() {
+    const pokemons_names_response = await this.pokemon_client.GetPokemonsNames();
+    this.ParsePokemonNamesResponse(pokemons_names_response);
+  }
+
+  /**
+     * sets the json object to the array of tasks
+     */
+  async SetArrayFromFile() {
+    try {
+      // get json object
+      const file_read_promise = await this.file_manager.ReadFromFileTasks();
+      // assing json object to tasks
+      this.tasks = file_read_promise;
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  /**
+     * gets string and checks what case it is
+     * @param {string} task_text string from input
+     */
+  async AddTask(task_text) {
+    // parse text to int
+    const pokemon_id = Number.parseInt(task_text, 10);
+    // check if text has commas
+    if (task_text.indexOf(',') > -1) {
+      const pokemos = await this.GetMultiplePokemons(task_text);
+      pokemos.forEach((pokemon) => this.InsertPokemonToArrayWrapperFunction(pokemon));
+    } else if (Number.isInteger(pokemon_id)) { // check if its number
+      const pokemon = await this.GetOnePokemon(pokemon_id);
+      this.InsertPokemonToArrayWrapperFunction(pokemon);
+    } else { // regular task
+      this.InsertResultToArray(task_text);
+    }
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+  }
+
+  /**
+     * calls the functions to insert pokemon correctly
+     * @param {Object} pokemon  pokemon object from api
+     */
+  InsertPokemonToArrayWrapperFunction(pokemon) {
+    if (pokemon.name !== 'Error') {
+      const pokemon_filtered = this.FilterPokemonAttributes(pokemon);
+      this.InsertResultToArray(pokemon_filtered);
+    } else { this.InsertResultToArray(pokemon); }
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+  }
+
+  /**
+     * removes a task from array
+     * @param {int} task_to_remove_id task location in array
+     */
+  RemoveTask(task_to_remove_id) {
+    const current_task_length = this.tasks.length;
```

Comment:
You could implement it in a more elegant way (filter for example).
It is not a common way to validate the id